### PR TITLE
added support for schemas to enable shallow testing

### PR DESCRIPTION
**Pull request by Markus-Ende**
Initial PR for issue #52 (option 1)
Maybe you can have a look, and we can discuss?

### CODE REVIEW

**Comment by nbeach** [comment]
The approach I've taken to accomplish a "shallow" rendering is to mock child components using `testBuilder.mock([SomeChild])`, rather than provide the `NO_ERRORS_SCHEMA`. The benefit I see here is that mocking the children allows you to assert:

-  the child components properly receive input bindings
-  the component under test responds to output events emitted by it's children
-  if/how instance methods on the child component class (accessed via `@ViewChild`) are called and what they return

When the `NO_ERRORS_SCHEMA` is used the children are essentially ignored, and (to my knowledge) you can't test any interactions with the child components. Are there scenarios you know of where `NO_ERRORS_SCHEMA` provide advantages vs the mocking or allows testing of the interactions in ways I'm not aware of?

Beyond the case at hand it seems reasonable that setting the schema could be useful though for some use cases.

**Comment by nbeach** [comment]
Looks solid! I would only ask you add a short blurb to the readme about the new method. 

Also I'm not sure why SuaceLabs keeps failing on this. I'll look into it.

**Comment by nbeach** [comment]
I found that Travis is (rightfully) not providing secret environment vars for external pull requests thus SauceLabs could not authenticate. I've disabled the browser tests for this scenario and (for now at least) will manually run the browser tests for external PRs before merging. 

TL;DR: I fixed the build process so this passes now 😄

**Comment by Markus-Ende** [comment]
> I would only ask you add a short blurb to the readme about the new method.

Done

**Comment by nbeach** [comment]
resolves #52

**Comment by nbeach** [comment]
I'm going to start getting the project into a release cadence of every Monday. So, this will be part of that next release for this coming week.
